spec/type_extensions/enum_spec.cr

```diff
@@ -2,21 +2,21 @@ require "../spec_helper"
 
 include LazyLoadHelpers
 
-describe "Enum" do
+describe Enum do
   it "check enum" do
     issue = IssueFactory.create
 
-    issue.status.enum.should eq(Issue::AvramStatus::Opened)
-    issue.role.enum.should eq(Issue::AvramRole::Issue)
+    issue.status.should eq(Issue::Status::Opened)
+    issue.role.should eq(Issue::Role::Issue)
   end
 
   it "update enum" do
     issue = IssueFactory.create
 
     updated_issue = Issue::SaveOperation.update!(issue, status: Issue::Status.new(:closed))
```

Comment:
Interesting. So we can still instantiate the enum like this? Can we also do `status: Issues::Status::Closed`?

Reply:
Yep, there's a bunch of ways to instantiate an enum for whatever reason

Reply:
So one way of querying that might have worked before but won't work this change is querying with symbols. For example `IssueQuery.new.status(:closed)` will not work. Even though you can do `Issue::Status.new(:closed)`, you can't actually do that through a method where a symbol is passed in.

Here's an example of something I was messing around with in Crystal playground to show the problem

```crystal
enum Color
  Red
  Blue
end

x = Color.new(:red)

def foo(x : Symbol)
  Color.new(x)
end

y = foo(:red)
```

`x` works but `y` causes a compilation error. This seems like it's a bug, not a feature

Reply:
In this case for `y`, `:red` isn't a `Symbol`, it's a `Color`. So if you do this:

```crystal
enum Color
  Red
  Blue
end

x = Color.new(:red)

def foo(x : Color)
  Color.new(x)
end

y = foo(:red)
```

It'll work. Crystal does the autocasting under the hood to say "You're asking for an Enum, and I see what looks like a Symbol, so I'll cast it.". If you specify Symbol though, crystal says "You're asking for a Symbol, but I need an Enum, so this doesn't work".

Reply:
Unfortunately I can't get it to work
<img width="597" alt="CleanShot 2021-07-10 at 13 41 52@2x" src="https://user-images.githubusercontent.com/17329408/125171876-a4b21900-e184-11eb-816a-e7e6b2d576ce.png">

Reply:
My guess is because it has to go through so many layers of methods

Reply:
Ah. That's fine I guess. This is still a huge improvement, and we can probably find another way later to make it work that way too.